src/OpenApi/src/Services/OpenApiDocumentService.cs

```diff
@@ -16,11 +18,20 @@ internal sealed class OpenApiDocumentService(
     [ServiceKey] string documentName,
     IApiDescriptionGroupCollectionProvider apiDescriptionGroupCollectionProvider,
     IHostEnvironment hostEnvironment,
-    IOptionsMonitor<OpenApiOptions> optionsMonitor)
+    IOptionsMonitor<OpenApiOptions> optionsMonitor,
+    IServiceProvider serviceProvider)
 {
     private readonly OpenApiOptions _options = optionsMonitor.Get(documentName);
 
-    public Task<OpenApiDocument> GetOpenApiDocumentAsync()
+    /// <summary>
+    /// Cache of <see cref="OpenApiOperationTransformerContext"/> instances keyed by the
+    /// `ApiDescription.ActionDescriptor.Id` of the associated operation. ActionDescriptor IDs
+    /// are unique within the lifetime of an application and serve as helpful associators between
+    /// operations, API description, and their respective transformer contexts.
+    /// </summary>
+    internal ConcurrentDictionary<string, OpenApiOperationTransformerContext> OperationTransformerContextCache = new();
```

Comment:
Maybe we could make this private (and readonly) and just expose a TryGet method?  It seems weird to give other types write access.

Reply:
I considered this but avoided it because I didn't think that much ceremony was necessary for an internal book-keeping field. I don't have strong opinions though and am happy to change.

Reply:
I'm not seriously concerned about other internal code modifying the cache, but I do think keeping a cache internal to a type is more conventional and conventional code is easier to read.